python/cudf/cudf/core/column/categorical.py

```diff
@@ -946,12 +946,14 @@ def unary_operator(self, unaryop: str):
         )
 
     def __setitem__(self, key, value):
-        to_add_categories = cudf.Index(value).difference(self.categories)
+        if cudf.utils.dtypes.is_scalar(
+            value
+        ) and cudf._lib.scalar._is_null_host_scalar(value):
+            to_add_categories = []
+        else:
+            to_add_categories = cudf.Index(value).difference(self.categories)
```

Comment:
Instead of actually storing the GPU-backed object here can we just store the integer length? It should reduce the peak memory usage here.

Reply:
Changed this to a counter